Source/Task/AsyncLib.cpp

```diff
@@ -438,10 +438,10 @@ static void CALLBACK CompletionCallbackForAsyncBlock(
     {
         asyncBlock->callback(asyncBlock);
     }
-    if (asyncBlock->waitEvent)
+    if (asyncBlock->internalHandle)
```

Comment:
Woah. I think this is all wrong (existing code).  Note that when we schedule the callback we pass in a state object, not an async block. This should attach to a state pointer, invoke with the state pointer's async block and then set the event in the state pointer.

Reply:
```c++
            hr = SubmitAsyncCallback(
                asyncBlock->queue,
                AsyncQueueCallbackType_Completion,
                asyncBlock,
                CompletionCallbackForAsyncBlock);
```
context is the AsyncBlock

---
In reply to: [193618839](https://github.com/Microsoft/libHttpClient/pull/186#discussion_r193618839) [](ancestors = 193618839)